src/views/Pipeline.vue

```diff
@@ -384,18 +384,27 @@ export default defineComponent({
         this.store.dispatch('job/fetchPendingJobs', {eComStoreId: this.getCurrentEComStore.productStoreId, viewSize:process.env.VUE_APP_VIEW_SIZE, viewIndex:0}).then(() => {
           if(event) event.target.complete();
           this.isRetrying = false;
+          emitter.off("selectedShop",this.refreshJobs);
+        }).catch(()=>{
+          this.isRetrying = false;
         });
       } else if(this.segmentSelected === 'running') {
         this.store.dispatch('job/fetchRunningJobs', {eComStoreId: this.getCurrentEComStore.productStoreId, viewSize:process.env.VUE_APP_VIEW_SIZE, viewIndex:0}).then(() => {
           if(event) event.target.complete();
           this.isRetrying = false;
+          emitter.off("selectedShop",this.refreshJobs);
```

Comment:
We can define this emitter.off outside the if else block, so that we don't need to redeclare in each if block.

Reply:
Done Sir